src/GaussianProcessEmulator.jl

```diff
@@ -66,11 +76,13 @@ struct GaussianProcess{FT<:AbstractFloat, GPM}
     "the Gaussian Process (GP) Regression model(s) that are fitted to the given input-data pairs"
     models::Vector
     "the singular value decomposition of obs_noise_cov, such that obs_noise_cov = decomposition.U * Diagonal(decomposition.S) * decomposition.Vt."
-    decomposition::Union{SVD, Nothing}
+    decomposition::Union{SVD, decomp_struct, Nothing}
```

Comment:
It's fine to create the decomposition structure here, 
might be nice to save everything in you `decomp_struct` format, then you can use `Union{decomp_struct, Nothing}` as the type.

You just need to add a constructor for `decomp_struct` for example
```julia
decomp_struct(svd::SVD) = decomp_struct(svd.V, svd.Vt, svd.S, size(svd.V)[0])
```